railgun/test/integration/controller_test.go

```diff
@@ -0,0 +1,59 @@
+//+build integration_tests
+
+package integration
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/kong/kubernetes-ingress-controller/railgun/manager"
+	"github.com/prometheus/common/expfmt"
+	"github.com/prometheus/common/model"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHealthEndpoint(t *testing.T) {
+	_ = proxyReady()
+	assert.Eventually(t, func() bool {
+		healthzURL := fmt.Sprintf("http://localhost:%v/healthz", manager.HealthzPort)
+		resp, err := httpc.Get(healthzURL)
+		if err != nil {
+			t.Logf("WARNING: error while waiting for %s: %v", healthzURL, err)
+			return false
+		}
+		defer resp.Body.Close()
+		return resp.StatusCode == http.StatusOK
+	}, ingressWait, waitTick)
+}
+
+func TestMetricsEndpoint(t *testing.T) {
+	if useLegacyKIC() {
+		t.Skip("metrics endpoint test does not apply to legacy KIC")
+	}
+	_ = proxyReady()
+	assert.Eventually(t, func() bool {
+		metricsURL := fmt.Sprintf("http://localhost:%v/metrics", manager.MetricsPort)
+		resp, err := httpc.Get(metricsURL)
+		if err != nil {
+			t.Logf("WARNING: error while waiting for %s: %v", metricsURL, err)
+			return false
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode == http.StatusOK {
+			decoder := expfmt.SampleDecoder{
+				Dec:  expfmt.NewDecoder(resp.Body, expfmt.FmtText),
+				Opts: &expfmt.DecodeOptions{},
+			}
+
+			var v model.Vector
+			if err := decoder.Decode(&v); err != nil {
+				t.Logf("decoder failed: %v", err)
+				return false
+			}
+
+			return len(v) > 0
+		}
+		return false
```

Comment:
nit - invert the condition so that the error flow gets indented and the "happy" flow remains unindented

https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

I should have spotted this when writing up my suggestion in the previous comment - it escaped my attention.

```suggestion
		if resp.StatusCode != http.StatusOK {
			return false
		}		
		decoder := expfmt.SampleDecoder{
			Dec:  expfmt.NewDecoder(resp.Body, expfmt.FmtText),
			Opts: &expfmt.DecodeOptions{},
		}

		var v model.Vector
		if err := decoder.Decode(&v); err != nil {
			t.Logf("decoder failed: %v", err)
			return false
		}

		return len(v) > 0
```